pkg/test/component_kinesis.go

```diff
@@ -2,63 +2,66 @@ package test
 
 import (
 	"fmt"
+	"github.com/applike/gosoline/pkg/cfg"
 	"github.com/aws/aws-sdk-go/service/kinesis"
-	"log"
 )
 
-type kinesisConfig struct {
-	Debug bool   `mapstructure:"debug"`
-	Host  string `mapstructure:"host"`
-	Port  int    `mapstructure:"port"`
+type kinesisSettings struct {
+	*mockSettings
+	Port int `cfg:"port"`
 }
 
-var kinesisConfigs map[string]*kinesisConfig
-var kinesisClients simpleCache
-
-func init() {
-	kinesisConfigs = map[string]*kinesisConfig{}
-	kinesisClients = simpleCache{}
+type kinesisComponent struct {
+	name     string
+	settings *kinesisSettings
+	clients  *simpleCache
+	runner   *dockerRunner
 }
 
-func ProvideKinesisClient(name string) *kinesis.Kinesis {
-	return kinesisClients.New(name, func() interface{} {
-
-		sess, err := getSession(kinesisConfigs[name].Host, kinesisConfigs[name].Port)
-
-		if err != nil {
-			logErr(err, "could not create kinesis client: %s")
-		}
-
-		return kinesis.New(sess)
-
-	}).(*kinesis.Kinesis)
-}
-
-func runKinesis(name string, config configInput) {
-	wait.Add(1)
-	go doRunKinesis(name, config)
+func (k *kinesisComponent) Boot(config cfg.Config, name string, settings *mockSettings, runner *dockerRunner) {
+	k.name = name
+	k.runner = runner
+	k.clients = &simpleCache{}
+	k.settings = &kinesisSettings{
+		mockSettings: settings,
+	}
+	key := fmt.Sprintf("mocks.%s", name)
+	config.UnmarshalKey(key, k.settings)
 }
 
-func doRunKinesis(name string, configMap configInput) {
-	defer wait.Done()
-	defer log.Printf("%s component of type %s is ready", name, "kinesis")
-
-	localConfig := &kinesisConfig{}
-	unmarshalConfig(configMap, localConfig)
-	kinesisConfigs[name] = localConfig
-
-	containerName := fmt.Sprintf("gosoline_test_kinesis_%s", name)
+func (k *kinesisComponent) Start() {
+	containerName := fmt.Sprintf("gosoline_test_kinesis_%s", k.name)
 
-	runContainer(containerName, ContainerConfig{
+	k.runner.Run(containerName, containerConfig{
 		Repository: "localstack/localstack",
 		Tag:        "0.10.7",
 		Env: []string{
 			"SERVICES=kinesis",
 		},
-		PortBindings: PortBinding{
-			"4568/tcp": fmt.Sprint(localConfig.Port),
+		PortBindings: portBinding{
+			"4568/tcp": fmt.Sprint(k.settings.Port),
 		},
-		HealthCheck: localstackHealthCheck(containerName),
-		PrintLogs:   localConfig.Debug,
+		HealthCheck: localstackHealthCheck(k.runner, containerName),
+		PrintLogs:   k.settings.Debug,
 	})
 }
+
+func (k *kinesisComponent) ProvideClients() *Clients {
+	return &Clients{
+		Kinesis: k.provideKinesisClient(),
+	}
+}
+
+func (k *kinesisComponent) provideKinesisClient() *kinesis.Kinesis {
+	return k.clients.New(k.name, func() interface{} {
+
+		sess, err := getAwsSession(k.settings.Host, k.settings.Port)
+
+		if err != nil {
+			panic(fmt.Errorf("could not create kinesis client %s: %w", k.name, err))
```

Comment:
in fact we couldn't create a aws session, no?

Reply:
I will panic in the getAwsSession function instead

✅